LineMsg[] equ
bool[] is_visible